core/src/test/java/org/apache/iceberg/rest/RESTCatalogAdapter.java

```diff
@@ -287,7 +288,23 @@ public RESTClient withAuthSession(AuthSession session) {
 
   @SuppressWarnings({"MethodLength", "checkstyle:CyclomaticComplexity"})
   public <T extends RESTResponse> T handleRequest(
-      Route route, Map<String, String> vars, Object body, Class<T> responseType) {
+      Route route,
+      Map<String, String> vars,
+      HTTPRequest httpRequest,
+      Class<T> responseType,
+      Consumer<Map<String, String>> responseHeaders) {
+    Object body = httpRequest.body();
+
+    if (responseHeaders != null) {
+      responseHeaders.accept(ImmutableMap.of("X-Request-Id", String.valueOf(System.nanoTime())));
+
+      // Set route specific response headers
+      switch (route) {
+        case CONFIG:
+          responseHeaders.accept(ImmutableMap.of("X-Iceberg-Version", IcebergBuild.version()));
+      }
```

Comment:
I realize this is public class in test package and shared by in many tests, want to leverage to ensure we cover the basic test of response header. Please let me know if there's better place to set response headers before features like idempotent etag is implemented.